### DISCUSSION TITLE
Followed all instructions for Jellyfin integration, but Authelia is asking for jwks option

### DISCUSSION THREAD

**Discussion by nothing2obvi** [authelia/authelia#7177] [Q&A] [answered]
How do I generate or acquire the values necessary for the jwks section for oidc?

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

To generate or acquire the values necessary for the `jwks` section for OpenID Connect (OIDC) in Authelia, you will need to create a set of JSON Web Keys (JWKs). Here are the steps to generate these keys:

1. **Generate a Private Key**: You can use a tool like OpenSSL to generate a private RSA key. Use the following command to generate a private key:
   ```
   openssl genrsa -out private.key 2048
   ```

2. **Create the JWK**: Convert the RSA private key to a JWK format. You can use a tool like `node-jose-tools` or an online converter. If using `node-jose-tools`, the command would be:
   ```
   jose newkey -s 256 -t oct -a HS256
   ```

3. **Extract the Public Key**: Extract the public key from the private key using OpenSSL:
   ```
   openssl rsa -in private.key -pubout -out public.key
   ```

4. **Convert the Public Key to JWK**: Similar to step 2, convert the public key to JWK format.

5. **Configure `jwks` in Authelia**: Once you have the JWK, you can configure it in the `jwks` section of your Authelia OIDC configuration. The configuration should include the `key_id`, `use`, `algorithm`, `key`, and optionally `certificate_chain` if applicable.

Here is an example of what the `jwks` section might look like:
```yaml
jwks:
  - key_id: 'example'
    use: 'sig'
    algorithm: 'RS256'
    key: |
      -----BEGIN PUBLIC KEY-----
      ...
      -----END PUBLIC KEY-----
    certificate_chain: |
      -----BEGIN CERTIFICATE-----
      ...
      -----END CERTIFICATE-----
      -----BEGIN CERTIFICATE-----
      ...
      -----END CERTIFICATE-----
```

This configuration includes manually configured JSON Web Keys, which is recommended to be used with the `jwks_uri` option for key rotation.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
The JWKS section which the Jellyfin guide links to contains information on generating them: https://www.authelia.com/configuration/identity-providers/openid-connect/provider/#key

**Comment by DJKatastrof** [comment]
So following the guide im only getting private.pem and public.pem.
The documentation states. Where do I find the .key file? Or is it enought with the private.pem? 

> identity_providers:
>   oidc:
>     jwks:
>       - key: {{ secret "/config/secrets/oidc/jwks/rsa.2048.key" | mindent 10 "|" | msquote }}
>

**Reply by Crowley723** [reply, marked as answer]
You'll want to use the private key file. (Private_2048_rsa.pem) the file name isn't as important as it being formated correctly, which should already be the case 

Since the public key is derived from the private, we only need the private key.
